test/planned_subvol_test.rb

```diff
@@ -207,4 +208,122 @@
     end
   end
 
+  context ".read_from_xml" do
```

Comment:
Elaborating what I mean with typical RSpec test vs set of examples.

This is how the output tests for the same functionality look in #200 
```
Y2Storage::ProposalSettings
  #read_product_features!
    [...] 
    sets #subvolumes based on the list at the 'subvolumes' feature
    [...]
    when reading the 'subvolumes' feature
      creates a PlannedSubvol for each subvolume in the list
      reads each 'path' value
      reads each 'copy_on_write' value
      uses true if 'copy_on_write' is omitted
      reads each 'archs' value as an array of strings
      uses nil if 'archs' is omitted
      for an empty list
        sets #subvolumes to an empty list
      if there is no 'subvolumes' feature
        sets #subvolumes to a fallback list
      if there are nil values in the list
        ignores the nil values
      if there are subvolumes without path 
        ignores those subvolumes
      when 'archs' contains just one name
        creates the planned subvolume if the architecture matches the name
        discards the subvolume if the architecture doesn't match
      when 'archs' contains just one name preceded by '!'
        discards the subvolume if the architecture matches the name
        also discards the subvolume if the architecture doesn't match
      when 'archs' contains a list of names
        creates the planned subvolume if the architecture matches any name
        discards the subvolume if none of the names matches the architecture
      when 'archs' contains names with and without '!'
        creates the subvolume if the positive names match and the negated don't
        discards the subvolume if both positive and negated names match
        discards the subvolume if none of the positive names match
      if there are several subvolumes with the same path
        if only one of them apply to the current architecture
          just creates the relevant planned subvolume
        if several of them apply to the current architecture
          just creates the last one to appear in the list
```
They describe the behavior serving as some kind of documentation to understand what's the purpose of the test. That's the goal of using RSpec and BDD.

This is the output of the equivalent test in this pull request.
```
Y2Storage::PlannedSubvol
  .create_from_xml
    Fully specified subvol
      has the correct path 
      is NoCOW
      is tolerant against whitespace in the archs list
      matches arch 'fake'
      matches arch 'ppc'
      does not match arch 'foo'
      does not match arch 'bar'
    Minimalistic subvol
      has the correct path
      is COW 
      is not arch-specific
      matches arch 'fake'
  .read_from_xml
    basic parsing
      returns a non-nil array
      removes architectures other than the current one
    a home subvolume
      exists
      has the correct path
      is COW
      is arch-specific
    a var/cache subvolume
      exists 
      has the correct path
      is COW
      is arch-specific
    a var/lib/mariadb subvolume
      exists
      has the correct path
      is NoCOW
      is arch-specific
    an arch-specific boot/grub2/s390x-emu subvolume
      exists on arch s390
      is COW
      is arch-specific 
    an arch-specific boot/grub2/i386-pc subvolume
      does not exist on s390
```
Things like "a var/cache volume" or "a var /lib/mariadb subvolume" look like random examples to me and give me no information about the intention of the test without reading the code (in the test or in the original class). Sure every example is there for a reason, but that's not clear from the output because it doesn't look like the typical RSpec one (a description of the behavior).

Thinking in terms of contexts and behavior really helps A LOT to find bugs while writing the tests (it forces you to think in width). If we just translate to RSpec the tests we did manually we are loosing the point and it's just a cumbersome syntax for writing traditional example-based unit tests.